File: Numeros Pseudo-Aleatorios/Park Miller/ParkMiller.cpp
/*
 * PRNG .......: Park-Miller
 * Ano ........: 1988
 * Autor ......: S. K. Park & K. W. Miller
*/

#include <iostream>
#include <iomanip>
#include <ctime>

unsigned int ParkMiller(unsigned int& X) {
    X = (16807 * X) % 2147483647;
    return X;
}

int main(void) {
  
    unsigned int Xn1, Xn2;
    int i;

    Xn1 = 1234;
    Xn2 = static_cast<unsigned int>(std::time(0));

    std::cout << "GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)" << std::endl;
    std::cout << "PSEUDORANDOM NUMBER GENERATOR (PRNG)" << std::endl;
    std::cout << "PADRAO: Park-Miller" << std::endl;
    std::cout << "==========================================" << std::endl;
    std::cout << std::endl;
    std::cout << "    Semente: 1234        Semente: TIME" << std::endl;
    std::cout << "    -------------        -------------" << std::endl;

    for (i = 1; i <= 10; i++) {
        std::cout << std::setw(17) << ParkMiller(Xn1);
        std::cout << std::setw(21) << ParkMiller(Xn2) << std::endl;
    }

    return 0;
}

File: Numeros Pseudo-Aleatorios/Park Miller/ParkMiller.dart
/**
 * PRNG .......: Park-Miller
 * Ano ........: 1988
 * Autor ......: S. K. Park & K. W. Miller
 */

import 'dart:io';
import 'dart:math';

int parkMiller(int x) {
  x = (16807 * x) % 2147483647;
  return x;
}

void main() {

  int x1 = 1234;
  int x2 = DateTime.now().millisecondsSinceEpoch;

  print('GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)');
  print('PSEUDORANDOM NUMBER GENERATOR (PRNG)');
  print('PADRAO: Park-Miller');
  print('==========================================');
  print('');
  print('    Semente: 1234        Semente: TIME');
  print('    -------------        -------------');

  for (int i = 1; i <= 10; i++) {
    x1 = parkMiller(x1);
    x2 = parkMiller(x2);
    print('${x1.toString().padLeft(17)}        ${x2.toString().padLeft(13)}');
  }

}
